src/Repository/QuizInstanceRepository.php

```diff
@@ -19,13 +27,26 @@ public function getQuestionsNumber(int $quizTemplateId): int
         return $dbStmt->fetchColumn();
     }
 
-    public function getResultsData(): array
+    /**
+     * Retrieves necessary data to display the results from database.
+     *
+     * @param Criteria $criteria
+     * @return SearchResult
+     */
+    public function getResultsData(Criteria $criteria): SearchResult
     {
         $sql = "SELECT quiz_instance.id AS quiz_instance_id, user.name, quiz_template.id AS quiz_template_id FROM quiz_instance 
                     JOIN user ON quiz_instance.user_id = user.id 
                     JOIN quiz_template ON quiz_instance.quiz_template_id = quiz_template.id";
+        $sql .= $criteria->toQuery();
         $dbStmt = $this->pdo->prepare($sql);
         $dbStmt->execute();
-        return $dbStmt->fetchAll();
+        $objects = $dbStmt->fetchAll();
+
+        $searchResult = new SearchResult();
+        $searchResult->setItems($objects);
+        $searchResult->setTotalCount($this->getNumberOfObjects($criteria));
```

Comment:
you don't really need to do 2 calls on the database here, you can use `count($objects)`